### PR TITLE
#1735 pane close button

### LINKED ISSUE #1735: Have a default way to open a pane
### Description

Currently, the user has to create a button to open a pane. The simplest solution for the user is to create a button, but this is not UX-friendly, so he also has to change the button's style. This requires a lot of code for a simple feature.

**Solution**

We should be able to provide a default way to open the pane with a great UI/UX.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Add an open button on closed Pane via `show_button` property
resolves #1735 

```
<|{show_pane}|pane|persistent|width=100px|show_button|
Pane content
|>
```

### CODE REVIEW

File: taipy/gui/viselements.json [hunk 1]
```diff
@@ -1883,6 +1883,12 @@
                         "type": "bool",
                         "default_value": "False",
                         "doc": "If False, the pane covers the page where it appeared and disappears if the user clicks in the page.<br/>If True, the pane appears next to the page. Note that the parent section of the pane must have the <i>flex</i> display mode set. See below for an example using the <code>persistent</code> property."
+                    },
+                    {
+                        "name": "show_button",
+                        "type": "bool",
+                        "default_value": "False",
+                        "doc": "if True, a button that allows to open the pane is shown when the pane is closed."
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                        "doc": "If True and when the pane is closed, a button allowing the pane to be opened is shown."
```

File: frontend/taipy-gui/src/components/Taipy/Pane.tsx [hunk 2]
```diff
@@ -133,6 +160,12 @@ const Pane = (props: PaneProps) => {
                 </>
             </Tooltip>
         </Drawer>
+    ) : showButton ? (
+        <Drawer variant="permanent" sx={buttonDrawerSx} anchor={anchor} open={true} className={getSuffixedClassNames(className, "-button")}>
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Pane.tsx:164]: Can you think of a way to find, in the code, all (or most of) those cases where computed classes were added to sub-components?

**Review by FabienLelaquais — DISMISSED** [review]
Looks great!

File: frontend/taipy-gui/src/components/Taipy/Pane.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Pane.tsx:164]: search for getSuffixedClassNames ?

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Pane.tsx:164]: Makes sense. Thx

**Review by FabienLelaquais — APPROVED** [review]
👌

**Comment by FlorianJacta** [comment]
Do you have a picture?

**Comment by FredLL-Avaiga** [comment]
> Do you have a picture?

image
for a pane anchored on right

**Review by dinhlongviolin1 — COMMENTED** [review]
LGTMMM

**Pull request merged**